9012-updating-userstorageuri-scheme

danarad05 opened the pull request:
scheme was changed from 'user_storage' to 'user-storage' as '\_' is not a valid char in scheme (according to [RFC 3986](https://tools.ietf.org/html/rfc3986#page-17))

#### What it does
Fixes second problem in #9012 where JDTUtils threw exception when receiving file changed event whose URI scheme was user_storage -which is an illegal scheme name (contains underscore).
Therefore changed `user_storage` to `user-storage` scheme name.

For first problem in #9012 see #9048

#### How to test
1. Pre-requisite:  #9048 is merged
2. Open in Theia java project
3. After load you can run or debug project
![image](https://user-images.githubusercontent.com/71069170/107147624-b4200280-6957-11eb-96f2-56f11a8225b5.png)

![image](https://user-images.githubusercontent.com/71069170/107147826-cc445180-6958-11eb-9364-e1ab13a64d62.png)

Note: 
for better behavior change this user setting to "internalConsole" (instead of "integratedTerminal"):
"java.debug.settings.console": "internalConsole".

#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

colin-grant-work commented:
It looks like the only way to address the problem. I've tested and confirmed that user storage is still being found correctly. It looks like perhaps a `sign-off` is missing in the commit  message (I believe that's why the Eclipse Foundation ECA check is failing), and the message itself is a bit long for a single line, but otherwise, this looks good. Hopefully there aren't too many explicit references to this URI in adopter code.

danarad05 commented:
> It looks like the only way to address the problem. I've tested and confirmed that user storage is still being found correctly - preferences etc. are read and updated as expected. It looks like perhaps a `sign-off` is missing in the commit message (I believe that's why the Eclipse Foundation ECA check is failing), and the message itself is a bit long for a single line, but otherwise, this looks good. Hopefully there aren't too many explicit references to this URI in adopter code.

Thanks @colin-grant-work for review. Fixed signoff and long commit message.
Thanks

amiramw commented:
I would like to merge this before the release if there is no objection
